signalfx/resource_signalfx_detector_test.go

```diff
@@ -249,6 +251,14 @@ func TestAccCreateUpdateDetector(t *testing.T) {
 	})
 }
 
+func waitBeforeTest(s *terraform.State) error {
```

Comment:
I'd rename this to make it more obvious that it does not wait before the configured tests of the step but before the plan/refresh phase that is done right after executing the tests:

```suggestion
func waitBeforeTestStepPlanRefresh(s *terraform.State) error {
```